lib/unix/git_unix.ml

```diff
@@ -186,17 +186,15 @@ module IO_FS = struct
     Lwt_pool.use mkdir_pool (fun () -> aux dirname)
 
   let list_files kind dir =
-    Lwt_pool.use openfile_pool (fun () ->
-        if Sys.file_exists dir then (
-          let s = Lwt_unix.files_of_directory dir in
-          let s = Lwt_stream.filter (fun s -> s <> "." && s <> "..") s in
-          let s = Lwt_stream.map (Filename.concat dir) s in
-          let s = Lwt_stream.filter kind s in
-          Lwt_stream.to_list s >>= fun l ->
-          Lwt.return l
-        ) else
-          Lwt.return_nil
-      )
+    if Sys.file_exists dir && Sys.is_directory dir then
+      let d = Sys.readdir dir in
+      let d = Array.to_list d in
+      let d = List.map (Filename.concat dir) d in
+      let d = List.filter kind d in
+      let d = List.sort String.compare d in
+      Lwt.return d
+    else
+      Lwt.return_nil
```

Comment:
I see, I don't think it deals with concurrent fd's but it seems like `Sys.readdir` itself wouldn't be interrupted. But the only thing that you're trading off here is performance so I think what we have here is fine.